scalameta/trees/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -364,6 +364,9 @@ object TreeSyntax {
         case t: Pat.Xml              =>
           val zipped = t.parts.zip(t.args).map{ case (part, arg) => s(part, "${", arg, "}") }
           m(SimplePattern, s(r(zipped), t.parts.last))
+        case Pat.Typed(lhs, rhs : Lit) =>
+          if (dialect.allowLiteralTypes) m(Pattern1, s(p(SimplePattern, lhs), kw(":"), " ", s(rhs)))
+          else throw new UnsupportedOperationException(s"$dialect doesn't support with types")
```

Comment:
On a second thought, let's remove prettyprinter-based validation here. I don't see how to do it concisely, because `Lit` can be used as a type in a lot of places.